### PR TITLE
Add permissible values to `FileTypeEnum` and `DataCategoryEnum` to accomodate workflow configuration `DataObject`s

### PR DESCRIPTION
**Pull request by brynnz22**
Resolves issue: https://github.com/microbiomedata/nmdc-schema/issues/1912

This PR (related to this initial PR that is closed: https://github.com/microbiomedata/berkeley-schema-fy24/pull/136 - we decided to go a different direction after Configuration discussions) allows `DataObject`s to include workflow configuration files, so they can be connected to their respective workflows via `has_input`

adds a`DataCategoryEnum` `permissible_value` of `workflow_parameter_data` to allow workflow parameter files as `has_input` for `WorfklowExecution`s. It also adds a `permissible_value` for the `FileTypeEnum` of `Configuration toml`.  Both new permissible values include EDAM mappings.

### CODE REVIEW

**Comment by brynnz22** [comment]
Unsure about how braod or specific the new permissible file for the `FileTypeEnum` needs to be. Wanted input from @SamuelPurvine

**Review by kheal — APPROVED** [review]
Looks good from my perspective.  Having the configuration tomls for the NOM, metabolomics, and lipidomics workflows is essential for reproducibility.  We can populate these `DataObject` sets programmatically alongside the workflow instances.  The mappings to edam are well matched to our needs, thanks!

**Review by turbomam — CHANGES_REQUESTED** [review]
I just reread this now and think this will be fine. Thanks for being flexible.

Please 
- modify the description of the new permissible value so that it mentions the things that it will interact with using their exact names from the schema
- add an example data file that shows the pattern by instantiating a Database

**Comment by turbomam** [comment]
Also please update the PR title to indicate what action was finally taken, adding a new permissible value

**Comment by kheal** [comment]
@brynnz22  - after talking with @SamuelPurvine, we think `workflow_parameter_data` is better than `workflow_parameter` (the latter sounds like we're capturing a single parameter), and fits better with the other permissible values on that Enum. 

The proteomics workflows will need a different set of permissible values for the `FileTypeEnum` (they are not tomls and there are more than one), so I think being more specific for that value for this specific use case is also valuable (`CoreMS Configuration toml`), but I am open to being overruled on that case. 

Generally speaking though, this approach is valuable and applicable for all the mass-spec 'omics, yay!

**Comment by brynnz22** [comment]
@turbomam I included you suggestions - updating the description for the `Configuration Toml` permissible value and including an example data set. 

I also changed the new `DataCateogryEnum` permissible value from `workflow_parameter` to `workflow_parameter_data` and @SamuelPurvine's request to be flexible to proteomics use cases.

File: src/schema/basic_slots.yaml [hunk 1]
```diff
@@ -890,6 +890,13 @@ enums:
       LC-DDA-MS/MS Raw Data:
         description: Liquid chromatographically separated MS1 and Data-Dependent MS2 binary instrument file
 
+      Configuration toml:
+        description: >-
+          Configuration toml file used by CoreMS in the NomAnlaysis and MetabolomicsAnalysis workflows to
```
Comment (turbomam) [inline src/schema/basic_slots.yaml:895]: - I don't thin CoreMS is defined anywhere in the schema. At a minimum, please provide a good see also
- please double check that the schema elements that you mention can be looked up verbatim in the docuemtnation or schema files
    - NomAnlaysis -> NomAnalysis

File: src/schema/basic_slots.yaml [hunk 2]
```diff
@@ -1044,4 +1051,8 @@ enums:
       instrument_data:
         description: Data generated by a data generation planned process
       processed_data:
-        description: Data generated by a workflow execution planned process
\ No newline at end of file
+        description: Data generated by a workflow execution planned process
```
Comment (turbomam) [inline src/schema/basic_slots.yaml:1054]: data generation -> DataGeneration 
workflow execution -> WorkflowExecution

**Review by turbomam — CHANGES_REQUESTED** [review]

**Review by turbomam — APPROVED** [review]
Good conversation with @brynnz22 , ultimately about something that she didn't actually add to the schema.

Will need some followup. see [comments in last commit](https://github.com/microbiomedata/berkeley-schema-fy24/pull/140/commits/81c63a7f7e14c2d61a6abf1b8006a7aef8f5fcd0).

**Pull request merged**